From 2576d706b0861f7a4b69ee20c2bf836b111fa2b6 Mon Sep 17 00:00:00 2001
From: chuan11x <chuangchuangx.liu@intel.com>
Date: Thu, 10 Oct 2019 01:59:16 -0400
Subject: [PATCH 2357/2367] Revert "media: intel-ipu4: ox03a10: correct sensor
 "

This reverts commit 17d3c14d2af3bf4de7d5ab3db47067945ec22b79

media: intel-ipu4: ox03a10: correct sensor register config

Change-Id: I96039fda5fc37f19319beef70305e668d9e634f1
Tracked-On: #JIIAP-842
Tracked-On: #PKT-2935
Signed-off-by: chuan11x <chuangchuangx.liu@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
Signed-off-by: Ong Yang Yang <yang.yangx.ong@intel.com>
---
 drivers/media/i2c/crlmodule/crl_ox03a10_common.h | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/media/i2c/crlmodule/crl_ox03a10_common.h b/drivers/media/i2c/crlmodule/crl_ox03a10_common.h
index 989c80f..a52609b 100644
--- a/drivers/media/i2c/crlmodule/crl_ox03a10_common.h
+++ b/drivers/media/i2c/crlmodule/crl_ox03a10_common.h
@@ -1678,7 +1678,11 @@ static struct crl_arithmetic_ops ox03a10_mirror_ops[] = {
 	{
 		.op = CRL_BITWISE_LSHIFT,
 		.operand.entity_val = 2,
-	}
+	},
+	{
+		.op = CRL_BITWISE_OR,
+		.operand.entity_val = 0x20,
+	},
 };
 
 static struct crl_dynamic_register_access ox03a10_h_flip_regs[] = {
@@ -1687,7 +1691,7 @@ static struct crl_dynamic_register_access ox03a10_h_flip_regs[] = {
 		.len = CRL_REG_LEN_08BIT | CRL_REG_READ_AND_UPDATE,
 		.ops_items = ARRAY_SIZE(ox03a10_mirror_ops),
 		.ops = ox03a10_mirror_ops,
-		.mask = 0x4,
+		.mask = 0x24,
 	},
 	{
 		.address = 0x3811,
-- 
2.7.4

